types/index.test-d.ts

```diff
@@ -91,6 +91,15 @@ async function main() {
   instance.get("/http2", (request, reply) => {
       reply.from("/", {
           method: "POST",
+          retryDelay: ({err, req, res, attempt, getDefaultDelay}: customRetryHandler) => {
```

Comment:
Might be a better test to not explicitly type the object here and let the inference prove that its all working?

```suggestion
          retryDelay: ({err, req, res, attempt, getDefaultDelay}) => {
```